### PR TITLE
More fun eval

### PR DESCRIPTION
**Pull request by darioizzo**
Adding all functions to the heyoka eval machinery.

### CODE REVIEW

File: test/expression.cpp [hunk 1]
```diff
@@ -59,35 +67,40 @@ void test_eval() {
...
         std::unordered_map<std::string, T> in{{"x", T(0.125)}};
         REQUIRE_THROWS(eval(ex, in));
     }
     // We test the implementation of all function.
     {
         std::unordered_map<std::string, T> in{{"x", T(0.125)}};
+        REQUIRE(eval(acos(x), in) == acos(T(0.125)));
+        //REQUIRE(eval(acosh(x+heyoka::expression(T(1.))), in) == acosh(T(1.125)));
+        REQUIRE(eval(asin(x), in) == asin(T(0.125)));
+        //REQUIRE(eval(asinh(x), in) == asinh(T(0.125)));
```
Comment (darioizzo) [inline test/expression.cpp:90]: These are commented out the tests fail (asinh already in double, 1e-17 diff) ..... I cannot see reasons that are not platform dependent ...

Comment (darioizzo) [reply test/expression.cpp:90]:
```/home/dario/Develop/heyoka/test/expression.cpp:92: FAILED:
  REQUIRE( eval(asinh(x), in) == asinh(T(0.125)) )
with expansion:
  0.1246767469 == 0.1246767469
```
difference is:  -1.38778e-17

**Comment by bluescarni** [comment]
@darioizzo I added a tolerance to the failing tests.

The only explanation I can think of is that when you write ``asinh(T(0.125))`` the compiler can compute the result at compile time, which it probably does in extended precision. Thus the exact result of the operation might differ slightly from the ``asinh()`` computed at runtime, because there are no requirements for these special function that they should always be correct up to the last digit (there are such requirements for simpler functions such as ``sqrt()`` and ``exp()`` I believe).

There was a similar report regarding the unit tests of mp++ at one point, and it turned out that it was an inconsistency between the value computed at compile time vs the value computed by the library function:

https://github.com/bluescarni/mppp/issues/132

**Comment by darioizzo** [comment]
thanks @bluescarni  I will use the tol then for the remaining ones when needed!

**Comment by darioizzo** [comment]
@bluescarni good to go

**Pull request merged**